src/main/java/spoon/support/visitor/equals/EqualsChecker.java

```diff
@@ -109,6 +110,15 @@ public void scanCtModifiable(CtModifiable m) {
 		super.scanCtModifiable(m);
 	}
 
+	@Override
+	public void scanCtCodeSnippet(CtCodeSnippet snippet) {
+		final CtCodeSnippet peek = (CtCodeSnippet) this.other;
+		if (!snippet.getValue().equals(peek.getValue())) {
+			setNotEqual(CtRole.SNIPPET);
```

Comment:
@slarse This method throws an [exception](https://github.com/INRIA/spoon/pull/4024/files#diff-d991c6c34e1ca95a4c24d3ddec98465f7517bc76a50cbcf04a0f0c866fffdf6eR56) eventually so I am confused why it doesn't exit the JVM.

Reply:
`EqualsChecker.scan()` is called within a try/catch: https://github.com/INRIA/spoon/blob/66db36cc8fb4e872597c51699e8aab1227a2542a/src/main/java/spoon/support/visitor/equals/EqualsVisitor.java#L38-L48